ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionManager.java

```diff
@@ -39,9 +39,26 @@ public void remove(Uid id)
 		_allActions.remove(id);
 	}
 
-    public int getNumberOfInflightTransactions()
+	/**
+	 * <p>
+	 *     This method works out the number of in-flight transactions currently in
+	 *     the system.
+	 * </p>
+	 * <p>
+	 *     Note: The definition of in-flight (AKA active) transactions, can be found
+	 *     in the OMG OTS specification: [active is defined as] the state of a
+	 *     transaction when processing is in progress and completion of the
+	 *     transaction has not yet commenced.
+	 * </p>
+	 *
+	 * @return the number of in-flight transactions currently in the system
+	 */
+	  public int getNumberOfInflightTransactions()
     {
-        return _allActions.size();
+        return (int) _allActions.entrySet()
+          .parallelStream()
+          .filter(entry -> entry.getValue().status() == ActionStatus.RUNNING)
```

Comment:
I want to point out that this specifically does not include transactions that are CREATED but not yet RUNNING (in case that is important to consider)

Reply:
I don't think we should track CREATED BasicActions as in-flight transactions as that status describes BasicActions that haven't been started yet (i.e. they are not active).

Reply:
Do we want to include those in the stat though?

Reply:
I wouldn't. It doesn't seem to me that counting CREATED BasicAction would add any information to Naryana's stats. Please, let me know if I'm missing something here.

Reply:
It depends what we are wanting to use this method for. If (for example) we want to know when all BasicActions that are in the system are no longer able to be "used" (pre-prepare/abort) then I think we should count them.

I would assume that (at least most, not certain on recovery) CREATED BasicAction's typically becoming RUNNING soon after but might not be RUNNING yet precisely when this method is called.

Reply:
The TM may crash after returning the stat in which case the CREATED transaction will never manifest as a running transaction.

And the stats are always a snapshot of the current state of the system so they are always going to be stale from the callers point of view.

Reply:
But a new stat seems like it will satisfy all of our viewpoints.

Reply:
I really like the idea of a new stat. I will create a new JBTM to track that effort. Thanks for the suggestion Michael

Reply:
https://issues.redhat.com/browse/JBTM-3903

Reply:
This increases the the time spent in the optimised [for throughput] path. Can we make JBTM-3903 dependent on https://issues.redhat.com/browse/JBTM-3901?

Reply:
I think it is worthwhile including my last comment in the JIRA so that whoever fixes it will take it into consideration.

Reply:
> I want to point out that this specifically does not include transactions that are CREATED but not yet RUNNING (in case that is important to consider)

@tomjenkinson Manuel created a new issue tracker for [this requirement](https://issues.redhat.com/browse/JBTM-3903) as a new statistic since in a crash scenario a CREATED transaction will never manifest as a running transaction. What priority do you want, I would suggest Optional (not that that would reduce its probability of being implemented but it's more reflective of its urgency), it's currently marked as Major.

Reply:
Optional sounds good to me. I modified the JBTM accordingly

Reply:
I will also make JBTM-3903 depend on JBTM-3901. Thanks Michael